src/backend/vulkan/src/lib.rs

```diff
@@ -1424,7 +1665,8 @@ pub struct RawDevice {
     features: Features,
     instance: Arc<RawInstance>,
     extension_fns: DeviceExtensionFunctions,
-    maintenance_level: u8,
+    /// Indicates this device is running on Vulkan 1.0 with no maintenance extensions.
+    is_core_1_0: bool,
```

Comment:
it seems confusingly general, while the real purpose is to check for Y-flip. We should make it either more versatile (to allow other pieces to rely on it), like the `maintenance_level` was, or more specific, i.e. `supports_y_flip: bool`

Reply:
I'll do `supports_y_flip: bool`. Is there a case in `map_viewport` where `flip_y` can be true but `shift_y` isn't? I suspect I've added a bug since I don't think I understand this bit of code.

Is it that vulkan 1.0 (+ amd negative viewport) didn't require shifting the viewport rect to flip it?

Reply:
It's the AMD extension that didn't require the shifting, yeah, but the normal stuff (i.e. maintenance or vk1.1) does need the shift.

Reply:
I've named it `flip_y_requires_shift` and added a comment to the field.